Content.Server/CombatMode/Pacification/PacificationSystem.cs

```diff
@@ -0,0 +1,53 @@
+using Content.Server.Fluids.Components;
+using Content.Server.Nutrition.Components;
+using Content.Server.Nutrition.EntitySystems;
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.CombatMode.Pacification;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Popups;
+using Content.Shared.Slippery;
+using Content.Shared.Throwing;
+
+namespace Content.Server.CombatMode.Pacification;
+
+public sealed class PacificationSystem : SharedPacificationSystem
+{
+    [Dependency] private   readonly SharedPopupSystem _popup = default!;
+    [Dependency] private   readonly OpenableSystem _openable = default!;
+    [Dependency] private   readonly SolutionContainerSystem _solutionContainerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PacifiedComponent, BeforeThrowEvent>(OnBeforeThrow);
+    }
+
+    private void OnBeforeThrow(EntityUid playerUid, PacifiedComponent component, BeforeThrowEvent args)
+    {
+        var thrownItem = args.ItemUid;
+        var itemName = Identity.Entity(thrownItem, EntityManager);
+        // Check whether the item being thrown is excluded by the blacklist:
+        if (component.ThrowBlacklist.IsValid(thrownItem, EntityManager))
```

Comment:
i think this should use a broadcast event for different systems to add things to, but check with a maintainer first

Reply:
That makes sense.

Reply:
I’ve started rewriting the logic to use events instead of a blacklist in the component, that makes a lot more sense.